irl_benchmark/rl/model/maze_world.py

```diff
@@ -0,0 +1,190 @@
+                            if state[num_rewards + a] != 0:
+                                rews_where = self.maze_env.rews_where
+                                rewards[s, a] += float(
+                                    self.maze_env.map_rewards[rews_where[0][a], \
+                                    rews_where[1][a]]) * (1 - RANDOM_QUIT_CHANCE)
+                        else:
+                            if reward_wrapper is None:
+                                rewards[s, a] = 0
+                            else:
+                                rewards[s, a] = wrapped_reward
+                    continue
+
+                success_prob = (1 - RANDOM_QUIT_CHANCE)**(len(path) - 1)
+                if return_transitions:
+                    new_state = get_next_state(state, a, num_rewards)
+                    new_s = self.state_to_index(new_state)
+                    add_transition(s, a, new_s, success_prob)
+                    add_transition(s, a, absorbing_s, 1. - success_prob)
+
+                if return_rewards:
+                    if reward_wrapper is None:
+                        if state[num_rewards + a] == 0:
+                            # if reward is already collected at this field:
+                            rew_value = 0
+                        else:
+                            rews_where = self.maze_env.rews_where
+                            rew_value = float(self.maze_env.map_rewards[
+                                rews_where[0][a], rews_where[1][a]])
+
+                        possible_distances = np.arange(1, len(path))
+                        prob_getting_to_distance = (
+                            1 - RANDOM_QUIT_CHANCE)**possible_distances
+                        prob_stopping_at_distance = np.ones_like(
+                            possible_distances, dtype=np.float32)
+                        prob_stopping_at_distance[:-1] = RANDOM_QUIT_CHANCE
+                        expected_walking_distance = np.sum(
+                            possible_distances * prob_getting_to_distance *
+                            prob_stopping_at_distance)
+                        weighted_reward = expected_walking_distance * REWARD_MOVE + success_prob * rew_value
+
+                        rewards[s, a] = weighted_reward
+                    else:
+                        rewards[s, a] = wrapped_reward
+
+        for a in range(n_actions):
+            if return_transitions:
+                add_transition(absorbing_s, a, absorbing_s, 1.)
+            if return_rewards:
+                rewards[absorbing_s, a] = 0
+
+        if return_transitions:
+            coords = np.array([
+                coords_trans_state, coords_trans_action,
+                coords_trans_next_state
+            ])
+            transitions = sparse.COO(coords, trans_data)
+
+        if return_transitions:
+            if return_rewards:
+                return transitions, rewards
+            return transitions
+        return rewards
```

Comment:
I tested this quite extensively, should be correct. I didn't add it to the pytests, since it runs for 10-20 seconds. We could maybe introduce the flag for long tests again.